script.js

```diff
@@ -1,4 +1,165 @@
+
+var rollDiceForPlayer = function () {
+  console.log("Control Flow: Start of rollDiceForPlayer()");
+  for (let i = 0; i < 2; i += 1) {
+    currentPlayerRolls.push(rollDice());
+  }
+  console.log(
+    "rollDiceForPlayer changes, currentPlayerRolls: " + currentPlayerRolls
+  );
+  return (
+    "Welcome, Player " +
+    currentPlayer +
+    "!<br><br>You rolled: <br>Dice 1: " +
+    currentPlayerRolls[0] +
+    " | Dice 2: " +
+    currentPlayerRolls[1] +
+    ".<br><br>Sweet! Now input either '1' or '2' to choose the corresponding dice to be used as the first digit of your final value."
+  );
+};
+
+var getPlayerScore = function (playerInput) {
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(
+      "Control Flow: Input validation, invalid input... INPUT NOT 1 OR 2"
+    );
+    return (
+      "Invalid input! Please only input '1' or '2' to choose which dice to use as the first digit.<br><br>Your dice rolls are:<br>Dice 1: " +
+      currentPlayerRolls[0] +
+      " | Dice 2: " +
+      currentPlayerRolls[1] +
+      "."
+    );
+  }
+  // input == 1
+  if (playerInput == 1) {
+    console.log("Control Flow: input == 1");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  // input == 2
+  if (playerInput == 2) {
+    console.log("Control Flow: input == 2");
+    var playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+  // store playerScore in array
+  allPlayersScore.push(playerScore);
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return (
+    "Player " + currentPlayer + ", your chosen value is: '" + playerScore + "'."
+  );
+};
+
+var comparePlayerScores = function () {
+  compareMessage =
+    "Player 1 Score: " +
+    allPlayersScore[0] +
+    "<br>Player 2 Score: " +
+    allPlayersScore[1];
+  // player 1 wins
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>Player 1 wins!";
+  }
+  // player 2 wins
+  if (allPlayersScore[0] < allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>Player 2 wins!";
+  }
+  // tie
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>It's a tie!";
+  }
+  return compareMessage;
+};
+
+var resetGame = function () {
+  currentPlayer = 1;
+  gameState = GAME_STATE_DICE_ROLL;
+  allPlayersScore = [];
+  return "<br><br>Click submit to play again!";
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log("Check game state on submit click: " + gameState);
+  console.log("Checking currentPlayer on submit click: " + currentPlayer);
+  var outputMessage = "";
+  if (gameState == GAME_STATE_DICE_ROLL) {
+    console.log("Control Flow: gameState == GAME_STATE_DICE_ROLL");
+
+    // display dice numbers rolled as output message
+    outputMessage = rollDiceForPlayer();
```

Comment:
One thing I noticed is that your helper functions tend to return strings for your output message. But if you take a look at your function naming, it does not seem to convey that meaning. 
As a reader of your code, I would expect rollDiceForPlayer to only roll the dice and maybe store the dice value in some array. I would not expect it to return me an entire string for the output message.
I'll talk more about this in class today